src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -630,6 +629,26 @@ namespace winrt::TerminalApp::implementation
         });
     }
 
+    // Method Description:
+    // - Get the combined taskbar state for the tab. This is the combination of
+    //   all the states of all our panes. Taskbar states are given a priority
+    //   based on the rules in:
+    //   https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-itaskbarlist3-setprogressstate
+    //   under "How the Taskbar Button Chooses the Progress Indicator for a
+    //   Group"
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - A TaskbarState object representing the combined taskbar state and
+    //   progress percentage of all our panes.
+    winrt::TerminalApp::TaskbarState TerminalTab::GetCombinedTaskbarState() const
+    {
+        std::vector<winrt::TerminalApp::TaskbarState> states;
+        _rootPane->CollectTaskbarStates(states);
+        std::sort(states.begin(), states.end(), TerminalApp::implementation::TaskbarState::ComparePriority);
+        return states.empty() ? winrt::make<winrt::TerminalApp::implementation::TaskbarState>() : states[0];

```

Comment:
1. Can `states` ever be empty? Wouldn't the tab not exist if that was the case?
2. Check out [`std::min_element`](https://en.cppreference.com/w/cpp/algorithm/min_element)

Reply:
1. Theoretically no, I was trying to be defensive about this one
2. nice